File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/wordlist/presentation/WordViewMapper.java
package com.przemyslawlusnia.vocabularycreator.wordlist.presentation;

import com.przemyslawlusnia.vocabularycreator.wordlist.domain.WordDomainModel;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class WordViewMapper {

  public WordViewModel mapToWordViewModel(WordDomainModel wordDomainModel) {
    return new WordViewModel(
        wordDomainModel.getWord(),
        wordDomainModel.getTranslation(),
        wordDomainModel.getType(),
        false);
  }

  public List<WordViewModel> mapToWordViewModels(List<WordDomainModel> wordDomainModels) {
    if (wordDomainModels == null) {
      return Collections.emptyList();
    }
    List<WordViewModel> result = new ArrayList<>(wordDomainModels.size());
    for (WordDomainModel word : wordDomainModels) {
      WordViewModel viewWord = new WordViewModel(
          word.getWord(),
          word.getTranslation(),
          word.getType(),
          false);
      result.add(viewWord);
    }
    return result;
  }

  public WordDomainModel mapToWordDomainModel(WordViewModel wordViewModel) {
    return new WordDomainModel(wordViewModel.getWord(), wordViewModel.getTranslation(), wordViewModel.getType());
  }

  public List<WordDomainModel> mapToWordDomainModels(List<WordViewModel> wordViewModels) {
    if (wordViewModels == null) {
      return Collections.emptyList();
    }
    List<WordDomainModel> result = new ArrayList<>(wordViewModels.size());
    for (WordViewModel word : wordViewModels) {
      result.add(new WordDomainModel(word.getWord(), word.getTranslation(), word.getType()));
    }
    return result;
  }

}

File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/wordlist/presentation/WordViewMapper.kt
package com.przemyslawlusnia.vocabularycreator.wordlist.presentation

import com.przemyslawlusnia.vocabularycreator.wordlist.domain.WordDomainModel

class WordViewMapper {

    fun mapToWordViewModel(wordDomainModel: WordDomainModel): WordViewModel {
        return WordViewModel(
                wordDomainModel.word,
                wordDomainModel.translation,
                wordDomainModel.type,
                false)
    }

    fun mapToWordViewModels(wordDomainModels: List<WordDomainModel>): List<WordViewModel> {
        return wordDomainModels.map {
            WordViewModel(
                    it.word,
                    it.translation,
                    it.type,
                    false)
        }
    }

    fun mapToWordDomainModel(wordViewModel: WordViewModel): WordDomainModel {
        return WordDomainModel(
                wordViewModel.word,
                wordViewModel.translation,
                wordViewModel.type)
    }

    fun mapToWordDomainModels(wordViewModels: List<WordViewModel>): List<WordDomainModel> {
        return wordViewModels.map { WordDomainModel(it.word, it.translation, it.type) }
    }

}
